### ISSUE TITLE
F1@M and F1@O

### ISSUE THREAD

**Issue by lightcome** [Chen-Wang-CUHK/ExHiRD-DKG#3]
Thank you for your work. Can you give an example of how F1@M and F1@O are calculated? And why not use F1@O as the evaluation index?

**Comment by lightcome** [comment]
For example, the real keywords are DCE MRI; cluster analysis; classification

The key words of prediction are: MRI DCE; cancer area classification, its F1 @ m and F1 @ o

What is not clear is whether the internal order of keywords will affect the evaluation, and how F1 @ m and F1 @ o are truncated. Compared with this, F1 @ m truncates the real keywords to "DCE MRI; cluster analysis" and discards "DCE MRI; cluster analysis" and then evaluate?

**Comment by Chen-Wang-CUHK** [comment]
> Thank you for your work. Can you give an example of how F1@M and F1@O are calculated? And why not use F1@O as the evaluation index?

In our paper, we use F1@5 and F1@M as the evaluation metrics. We do not use F1@O because the metric needs to fine-tune a hyperparameter on the validation datasets. As for F1@5 and F1@M, they have different methods to compute the precision score.  There are two examples:

Case 1 (6 predicted keyphrases (KPs) after the post-processing): KP1, KP2, KP3, KP4, KP5, KP6
Suppose 3 are correct, then the precisions are computed by: P@5=3/5=0.6; P@M=3/6=0.5.

Case 2 (3 predicted keyphrases (KPs) after the post-processing): KP1, KP2, KP3
Suppose 3 are correct, then the precisions are computed by: P@5=3/5=0.6; P@M=3/3=1.0.
